libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -446,7 +463,15 @@ void AP_Camera::update_trigger()
         DataFlash_Class *df = DataFlash_Class::instance();
         if (df != nullptr) {
             if (df->should_log(log_camera_bit)) {
-                df->Log_Write_Camera(ahrs, current_loc);
+                uint64_t timestamp = 0;
+                if (_feedback_sem && _feedback_sem->take_nonblocking()) {
+                    timestamp = _feedback_timestamp_us;
```

Comment:
`_feedback_timestamp_us` needs to be cleared back to 0, or you will always get an old timestamp if the semaphore wasn't taken.